### ISSUE TITLE
Error parsing `get_api_test_result` data

### ISSUE THREAD

**Issue by trondhindenes** [DataDog/datadog-api-client-python#1035]
**Describe the bug**
When checking the status of a synthetics test, I'm getting the following error (if the test has completed and failed):

```
datadog_api_client.exceptions.ApiTypeError: Invalid type for variable 'date'. Required value type is dict and passed type was str at ['received_data']['result']['response_headers']['date']
```

**To Reproduce**
this only happens if the configured test has completed and the test failed, for example if the api returns a 500 while the test assertion required a 200 status.

**Expected behavior**
the call should not fail

**Screenshots**

**Environment and Versions (please complete the following information):**
`datadog-api-client==1.12` on python 3.9

**Additional context**

**Comment by therve** [comment]
Thanks for your report, should be fixed by #1036

**Comment by therve** [comment]
Fix will be in the next release, sorry for the troubles.
